src/nemos/regularizer.py

```diff
@@ -621,7 +459,42 @@ def _check_mask(mask: jnp.ndarray):
                 f"Data type {mask.dtype} provided instead!"
             )
 
-    def _get_proximal_operator(
+    def _penalization(
+        self, params: Tuple[DESIGN_INPUT_TYPE, jnp.ndarray]
+    ) -> jnp.ndarray:
+        """
+        Calculate the penalization.
```

Comment:
this is private, but for developers we should point out to an equatio/article. In the proximal_operator I have the reference to the group lasso paper

        Yuan, Ming, and Yi Lin. "Model selection and estimation in regression with grouped variables."
        Journal of the Royal Statistical Society Series B: Statistical Methodology 68.1 (2006): 49-67.

The penalized loss is 
$$\\text{loss}(\beta_1,...,\beta_g) + \alpha \cdot \sum _{j=1...,g} \sqrt{\dim(\beta_j)} || \beta_j||_2$$

where $g$ is the number of groups, $\dim(\cdot)$ is the dimension of the vector, i.e. the number of coefficient in each $\beta_j$, and $||\cdot||_2$ is the euclidean norm.

Reply:
okay sure, will add